test/SILOptimizer/remove_unused_global_vars.swift

```diff
@@ -0,0 +1,28 @@
+// RUN: %target-swift-frontend -O -emit-sil %s | %FileCheck %s
+
+// CHECK-NOT: sil_global private [let] {{.*}}unused1{{.*}}
+private let unused1 = 0
+// CHECK-NOT: sil_global private {{.*}}unused2{{.*}}
+private var unused2 = 42
```

Comment:
Can you add some tests where the global has a non-trivial initial value that doesn't get optimized down to a constant? We should be able to eliminate the accessor function as well.

Reply:
Added the tests you suggested.